WeakAuras/Prototypes.lua

```diff
@@ -2972,7 +2974,7 @@ Private.event_prototypes = {
 
       return ret
     end,
-    statesParameter = "unit",
+    statesParameter = "one",
```

Comment:
Since `CURRENT_SPELL_CAST_CHANGED` is not a `UNIT` event, the triggerFunc wasn't getting called for that event when `statesParameter` was set to `"unit"`. It looks like this might also change the key used in the `allstates` table later on, from e.g. `"player"` to `""`? I wasn't sure and would appreciate feedback if there's a better way to add a non-unit event.

Reply:
This will break the power trigger for unit=boss/nameplate/group/...

Reply:
I was worried about something like that. The reason I changed it is that `CURRENT_SPELL_CAST_CHANGED`'s first argument is sometimes `false`, so with `statesParameter` set to `unit`, this check in RunTriggerFunc fails, and the trigger func is never run:

https://github.com/WeakAuras/WeakAuras2/blob/d0f975fe0efc30c04a37a24b4e55c263a3855c7a/WeakAuras/GenericTrigger.lua#L587

Reply:
Changed it to `if arg1 ~= nil` and reverted the power trigger's `statesParameter` to `unit`

Reply:
I don't like this solution, if trigger expect an event with a unit as arg1, it needs to receive an event with a unit as arg1.
You can take the cast trigger as an example with WeakAuras.WatchForCastLatency or WeakAuras.WatchForNameplateTargetChange and WA_UNIT_TARGET_NAME_PLATE internal event

Reply:
Thanks for the feedback, I've refactored it to use a helper function/event following that pattern.